xclim/subset.py

```diff
@@ -924,7 +924,9 @@ def subset_gridpoint(
     # Subset with specific start_dates and end_dates
     >>> prSub = subset.subset_time(ds.pr,lon=-75,lat=45, start_date='1990-03-13',end_date='1990-08-17')
     """
-    # check if trying to subset lon and lat
+    if lat is None or lon is None:
+        raise ValueError("Insufficient coordinates provided to locate grid point(s).")
+
     if lat is not None and lon is not None:
```

Comment:
I think if we remove this level of condition, complexity will go down.

Reply:
Right.